### PR TITLE
Bluetooth: controller: Refactored llcp_req/ack test to helper function

### PR DESCRIPTION
**Pull request by alexsven**
Moved test of llcp_rec to separate helper function in ull.c

Signed-off-by: Alexander Svensen <alsv@nordicsemi.no>

### CODE REVIEW

**Comment by jfischer-phytec-iot** [comment]
@alexsven Something went wrong, please rebase on master.

File: boards/arm/atsamd20_xpro/doc/atsamd20_xpro.rst [hunk 1]
```diff
@@ -134,7 +134,7 @@ References
 .. target-notes::
 
 .. _Microchip website:
-    www.microchip.com/DevelopmentTools/ProductDetails.aspx?PartNO=ATSAMD20-XPRO
+    https://www.microchip.com/DevelopmentTools/ProductDetails.aspx?PartNO=ATSAMD20-XPRO
```
Comment (dbkinder) [inline boards/arm/atsamd20_xpro/doc/atsamd20_xpro.rst]: You may need to rebase. You've picked up a stray commit.

**Review by dbkinder — DISMISSED** [review]

**Review by finikorg — DISMISSED** [review]
Please merge those patches together

File: boards/arm/atsamd20_xpro/doc/atsamd20_xpro.rst (hunk 1 above)
Comment (alexsven) [reply boards/arm/atsamd20_xpro/doc/atsamd20_xpro.rst]: Done

File: subsys/bluetooth/controller/ll_sw/ull_internal.h [hunk 2]
```diff
@@ -14,6 +14,7 @@ static inline void ull_hdr_init(struct ull_hdr *hdr)
 	hdr->disabled_cb = hdr->disabled_param = NULL;
 }
 
+int is_bt_hci_cmd_disallowed(void *conn);
```
Comment (thoh-ot) [inline subsys/bluetooth/controller/ll_sw/ull_internal.h]: Rename the function ull_is_hci_cmd_disallowed, as this is only used within the ull and also matches the existing naming of other ull functions.

**Review by thoh-ot — CHANGES_REQUESTED** [review]

File: subsys/bluetooth/controller/ll_sw/ull_internal.h (hunk 2 above)
Comment (alexsven) [reply subsys/bluetooth/controller/ll_sw/ull_internal.h]: Done

**Comment by thoh-ot** [comment]
Please merge those patches together

**Comment by alexsven** [comment]
> Please merge those patches together
@thoh-ot: So there should never be more than one commit in a PR?

**Comment by thoh-ot** [comment]
> > Please merge those patches together
> > @thoh-ot: So there should never be more than one commit in a PR?

No that is not the case.

The issue is that commits to fixup a PR should be merged into the original commit they fix, such that we don't end up with a noisy history of commits that say "oh forgot to add this", "adding this because is't nicer", "had a typo here", etc. 

See it as the commit(s) in the PR is the final state of the code you wan't to present.

It's perfectly allowable to have multiple commits in a single PR. The only rule is that each commit should be able to stand alone (see https://docs.zephyrproject.org/latest/contribute/index.html for further details).

**Comment by alexsven** [comment]
@thoh-ot Ah, I see. Thank you for clarifying

**Review by thoh-ot — APPROVED** [review]

**Review by cvinayak — CHANGES_REQUESTED** [review]
Fix "llcp_rec"  to "llcp_req/ack" in the commit title and message.

File: subsys/bluetooth/controller/ll_sw/ull.c [hunk 3]
```diff
@@ -785,6 +786,22 @@ void ll_rx_mem_release(void **node_rx)
 	_rx_alloc(UINT8_MAX);
 }
 
+int ull_is_hci_cmd_disallowed(void *conn)
```
Comment (cvinayak) [inline subsys/bluetooth/controller/ll_sw/ull.c]: Move this to ull_conn.c and rename to ull_conn_is_hci_cmd_disallowed.

Comment (cvinayak) [reply subsys/bluetooth/controller/ll_sw/ull.c]: also return value is u8_t

File: subsys/bluetooth/controller/ll_sw/ull_internal.h [hunk 4]
```diff
@@ -14,6 +14,7 @@ static inline void ull_hdr_init(struct ull_hdr *hdr)
 	hdr->disabled_cb = hdr->disabled_param = NULL;
 }
 
+int ull_is_hci_cmd_disallowed(void *conn);
```
Comment (cvinayak) [inline subsys/bluetooth/controller/ll_sw/ull_internal.h]: Move to ull_conn_internal.h

File: subsys/bluetooth/controller/ll_sw/ull_conn.c [hunk 5]
```diff
@@ -230,13 +230,8 @@ u8_t ll_conn_update(u16_t handle, u8_t cmd, u8_t status, u16_t interval_min,
...
 
 	if (!cmd) {
-		if (conn->llcp_req != conn->llcp_ack) {
-			return BT_HCI_ERR_CMD_DISALLOWED;
-		}
 
-		conn->llcp_req++;
-		if (((conn->llcp_req - conn->llcp_ack) & 0x03) != 1) {
-			conn->llcp_req--;
+		if (ull_is_hci_cmd_disallowed(conn)) {
```
Comment (cvinayak) [inline subsys/bluetooth/controller/ll_sw/ull_conn.c]: assign the return value, check and return the value?

**Review by cvinayak — CHANGES_REQUESTED** [review]

**Comment by thoh-ot** [comment]
You still have "llcp_rec" in the title, it should be "llcp_req".

**Comment by carlescufi** [comment]
@alexsven 
`error: ‘ret’ undeclared (first use in this function)`

**Comment by alexsven** [comment]
> @alexsven
> `error: ‘ret’ undeclared (first use in this function)`

Fixed

**Review by carlescufi — CHANGES_REQUESTED** [review]

**Review by carlescufi — APPROVED** [review]

**Review by cvinayak — APPROVED** [review]
LGTM

**Pull request merged**